src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/DgmlWriter.cs

```diff
@@ -0,0 +1,320 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if DEBUG
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Net;
+
+namespace System.Text.RegularExpressions.Symbolic
+{
+    internal static class DgmlWriter<T> where T : notnull
+    {
+        /// <summary>Write the DFA or NFA in DGML format into the TextWriter.</summary>
+        /// <param name="matcher">The <see cref="SymbolicRegexMatcher"/> for the regular expression.</param>
+        /// <param name="writer">Writer to which the DGML is written.</param>
+        /// <param name="nfa">True to create an NFA instead of a DFA.</param>
+        /// <param name="addDotStar">True to prepend .*? onto the pattern (outside of the implicit root capture).</param>
+        /// <param name="reverse">If true, then unwind the regex backwards (and <paramref name="addDotStar"/> is ignored).</param>
+        /// <param name="maxStates">The approximate maximum number of states to include; less than or equal to 0 for no maximum.</param>
+        /// <param name="maxLabelLength">maximum length of labels in nodes anything over that length is indicated with .. </param>
+        public static void Write(
```

Comment:
Should all of this code be excluded from code coverage even when it is only build in debug?